cob_scan_unifier/src/scan_unifier_node.cpp

```diff
@@ -230,26 +233,26 @@ void scan_unifier_node::topicCallbackLaserScan(const sensor_msgs::LaserScan::Con
 sensor_msgs::LaserScan scan_unifier_node::unifieLaserScans()
 {
   sensor_msgs::LaserScan unified_scan = sensor_msgs::LaserScan();
-  std::vector<sensor_msgs::PointCloud> vec_cloud;
-  vec_cloud.assign(config_.number_input_scans, sensor_msgs::PointCloud());
+  std::vector<sensor_msgs::PointCloud2> temp_cloud;
+  temp_cloud.assign(config_.number_input_scans, sensor_msgs::PointCloud2());
+  std::vector<pcl::PointCloud<pcl::PointXYZI> > vec_cloud;
+  vec_cloud.assign(config_.number_input_scans, pcl::PointCloud<pcl::PointXYZI>());
 
   if(!vec_laser_struct_.empty())
   {
     ROS_DEBUG("start converting");
     for(int i=0; i < config_.number_input_scans; i++)
     {
-      vec_cloud.at(i).header.stamp = vec_laser_struct_.at(i).current_scan_msg.header.stamp;
+      temp_cloud.at(i).header.stamp = vec_laser_struct_.at(i).current_scan_msg.header.stamp;
       ROS_DEBUG_STREAM("Converting scans to point clouds at index: " << i << ", at time: " << vec_laser_struct_.at(i).current_scan_msg.header.stamp << " now: " << ros::Time::now());
       try
       {
-        listener_.waitForTransform("/base_link", vec_laser_struct_.at(i).current_scan_msg.header.frame_id,
-            vec_laser_struct_.at(i).current_scan_msg.header.stamp, ros::Duration(3.0));
-
         ROS_DEBUG("now project to point_cloud");
-        projector_.transformLaserScanToPointCloud("/base_link",vec_laser_struct_.at(i).current_scan_msg, vec_cloud.at(i), listener_);
+        projector_.transformLaserScanToPointCloud("base_link", vec_laser_struct_.at(i).current_scan_msg, temp_cloud.at(i), *p_tfBuffer.get());
+        pcl::fromROSMsg<pcl::PointXYZI>(temp_cloud.at(i), vec_cloud.at(i));
       }
-      catch(tf::TransformException ex){
-        ROS_ERROR("%s",ex.what());
+      catch(tf2::TransformException ex){
+        //ROS_ERROR("%s",ex.what());
```

Comment:
Because I get too much output. The thing is that underlying function `transformLaserScanToPointCloud` uses `ros::time::now()` and since if often needed "extrapolation in the future" there are errors. With tf is the same problem. So not to have full screen of errors I commented it out.

I know this is not optimal solution but I don't have better idea...